src/order/bounded_order.lean

```diff
@@ -666,6 +666,17 @@ instance [partial_order α] : partial_order (with_bot α) :=
   end,
   .. with_bot.preorder }
 
+lemma map_fn_le_iff {α β} [preorder α] [linear_order β] (f : α → β)
+  (a b : with_bot α) (mono_iff : ∀ {a b}, f a ≤ f b ↔ a ≤ b) :
+  a.map f ≤ b.map f ↔ a ≤ b :=
+begin
+  induction a using with_bot.rec_bot_coe,
+  { simp },
+  { induction b using with_bot.rec_bot_coe,
+    { simp [not_coe_le_bot a] },
+    { simpa using mono_iff } }
+end
```

Comment:
This might be worth restating with `strict_mono` or `monotone` somehow. @YaelDillies?

Reply:
I would rather make this a def about `order_embedding`. Also I'm surprised that you need `linear_order` for this.